### ISSUE TITLE
Error: Provider produced inconsistent result after apply

### ISSUE THREAD

**Issue by bibek-rauniyar** [microsoft/terraform-provider-azuredevops#279]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

I am getting the below error while creating the service connection for Kubernetes. 
I have already taken care of cyclic dependencies but also not able to resolve this issue.  Please let me know the workaround for this issue.
### Terraform (and Azure DevOps Provider) Version

Terraform v0.13.5

### Affected Resource(s)

* `azuredevops_XXXXX`

### Terraform Configuration Files
```hcl
locals {
  service_endpoint_name = "${var.common_name_prefix}-${var.name}-${var.environment}-sp-conn-terraform"
}
data "azuredevops_project" "main" {
  name = var.project_name
}
resource "azuredevops_serviceendpoint_kubernetes" "main" {
  project_id            = data.azuredevops_project.main.id
  service_endpoint_name = local.service_endpoint_name
  apiserver_url         = var.apiserver_url
  authorization_type    = var.authorization_type
  
  azure_subscription {
    subscription_id   = var.subscription_id
    subscription_name = var.subscription_name
    tenant_id         = var.tenant_id
    resourcegroup_id  = var.resourcegroup_id
    namespace         = var.namespace
    cluster_name      = var.cluster_name
    cluster_admin     = var.cluster_admin
  }
}

resource "time_sleep" "wait_120_seconds" {
  depends_on = [azuredevops_serviceendpoint_kubernetes.main]
  create_duration = "300s"
}

resource "azuredevops_resource_authorization" "auth" {
  project_id  = data.azuredevops_project.main.id
  resource_id = azuredevops_serviceendpoint_kubernetes.main.id
  authorized  = true
  depends_on = [time_sleep.wait_120_seconds]
}
```

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp
```

### Debug Output
```log
2021/01/25 20:34:06 [DEBUG] Azure Backend Response for https://oneappstoragedev.blob.core.windows.net/terraform-backend-state/stage%2Foneapp%2Fservice-connection%2Faks%2Fterraform.tfstate:
HTTP/1.1 201 Created
Content-Length: 0
Content-Md5: Pk2SFGLuOg+KESRzohE3lg==
Date: Mon, 25 Jan 2021 14:49:06 GMT
Etag: "0x8D8C1405E09EE87"
Last-Modified: Mon, 25 Jan 2021 14:49:07 GMT
Server: Windows-Azure-Blob/1.0 Microsoft-HTTPAPI/2.0
X-Ms-Request-Id: e60bab98-a01e-0027-7c29-f3c6c7000000
X-Ms-Request-Server-Encrypted: true
X-Ms-Version: 2018-11-09



Error: Provider produced inconsistent result after apply

2021/01/25 20:34:06 [DEBUG] Azure Backend Request:
HEAD /terraform-backend-state/stage%2Foneapp%2Fservice-connection%2Faks%2Fterraform.tfstate HTTP/1.1
Host: oneappstoragedev.blob.core.windows.net
User-Agent: Terraform/0.13.5
X-Ms-Date: Mon, 25 Jan 2021 14:49:06 GMT
X-Ms-Lease-Id: 54b836f0-9046-42b1-5d07-0db168e13993
X-Ms-Version: 2018-11-09


When applying changes to azuredevops_resource_authorization.auth, provider
"registry.terraform.io/microsoft/azuredevops" produced an unexpected new
value: Root resource was present, but now absent.

This is a bug in the provider, which should be reported in the provider's own
issue tracker.

2021/01/25 20:34:06 [DEBUG] Azure Backend Response for https://oneappstoragedev.blob.core.windows.net/terraform-backend-state/stage%2Foneapp%2Fservice-connection%2Faks%2Fterraform.tfstate:
HTTP/1.1 200 OK
Content-Length: 3908
Accept-Ranges: bytes
Content-Md5: Pk2SFGLuOg+KESRzohE3lg==
Content-Type: application/json
Date: Mon, 25 Jan 2021 14:49:06 GMT
Etag: "0x8D8C1405E09EE87"
Last-Modified: Mon, 25 Jan 2021 14:49:07 GMT
Server: Windows-Azure-Blob/1.0 Microsoft-HTTPAPI/2.0
X-Ms-Access-Tier: Hot
X-Ms-Access-Tier-Inferred: true
X-Ms-Blob-Type: BlockBlob
X-Ms-Creation-Time: Wed, 20 Jan 2021 11:22:26 GMT
X-Ms-Lease-Duration: infinite
X-Ms-Lease-State: leased
X-Ms-Lease-Status: locked
X-Ms-Meta-Terraformlockid: eyJJRCI6IjU0YjgzNmYwLTkwNDYtNDJiMS01ZDA3LTBkYjE2OGUxMzk5MyIsIk9wZXJhdGlvbiI6Ik9wZXJhdGlvblR5cGVBcHBseSIsIkluZm8iOiIiLCJXaG8iOiJiaWJla3Jvbml5YXJAQmliZWstUm9uaXlhci5sb2NhbCIsIlZlcnNpb24iOiIwLjEzLjUiLCJDcmVhdGVkIjoiMjAyMS0wMS0yNVQxNDo0Mzo0Ny4wNDU4MzNaIiwiUGF0aCI6InRlcnJhZm9ybS1iYWNrZW5kLXN0YXRlL3N0YWdlL29uZWFwcC9zZXJ2aWNlLWNvbm5lY3Rpb24vYWtzL3RlcnJhZm9ybS50ZnN0YXRlIn0=
X-Ms-Request-Id: e60babc4-a01e-0027-2529-f3c6c7000000
X-Ms-Server-Encrypted: true
X-Ms-Version: 2018-11-09
```

### Steps to Reproduce

1. `terraform apply`

### References

* #0000

**Comment by bibek-rauniyar** [comment]
Can anyone help me to solve this issue please?

**Comment by xuzhang3** [comment]
Hi @bibek-rauniyar This is a  know issue.  Previously, ADO won't check the service connection status which will ignore the creation error even we can find it in the service connection page.  Latest ADO(not released) will check the service connection status and make sure the service connection is created without error. #273

**Comment by xuzhang3** [comment]
Hi @bibek-rauniyar v0.1.2 has been released,  v0.1.2 should solve your problem
